tasks/buildsources.js

```diff
@@ -40,50 +53,66 @@ module.exports = function(grunt) {
 		}
 		scanDir(polyfillSourceFolder);
 
-		dirs.forEach(function(polyfillPath) {
+		var promises = dirs.map(function(polyfillPath) {
 			var config;
 			var configPath = path.join(polyfillPath, 'config.json');
 			var detectPath = path.join(polyfillPath, 'detect.js');
 			var polyfillSourcePath = path.join(polyfillPath, 'polyfill.js');
 			var featureName = polyfillPath.substr(polyfillSourceFolder.length+1).replace(/\//g, '.');
 
+			// Load the polyfill's configuration
 			try {
+				if (!fs.existsSync(configPath)) return;
+				config = JSON.parse(fs.readFileSync(configPath));
+				config.baseDir = path.relative(path.join(__dirname,'../polyfills'), polyfillPath);
+			} catch (e) {
+				throw {name:"Invalid config", message:"Unable to read config from "+configPath};
+			}
 
-				// Load the polyfill's configuration
-				try {
-					if (!fs.existsSync(configPath)) return;
-					config = JSON.parse(fs.readFileSync(configPath));
-					config.baseDir = path.relative(path.join(__dirname,'../polyfills'), polyfillPath);
-				} catch (e) {
-					throw {name:"Invalid config", message:"Unable to read config from "+configPath};
-				}
+			config.hasTests = fs.existsSync(path.join(polyfillPath, 'tests.js'));
 
-				config.hasTests = fs.existsSync(path.join(polyfillPath, 'tests.js'));
+			if (config.licence) {
+				throw 'Incorrect spelling of license property in '+featureName;
+			}
 
-				if (config.licence) {
-					throw 'Incorrect spelling of license property in '+featureName;
-				}
+			if (config.build && config.build.commonjs === true) {
+				config.rawSource = streamToString(browserify().add(polyfillSourcePath).bundle());
```

Comment:
Each polyfill is stored as a separate json structure on the filesystem, when a browser requests a set of polyfills the service will combine the necessary files together and return them as the response. Because of this approach, we can only run computations such as browserify for each individual polyfill.